engine/src/main/java/org/apache/apex/stram/DeployRequest.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.stram;
+
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+
+import com.datatorrent.stram.util.AbstractWritableAdapter;
+
+public class DeployRequest
+{
+  private EventGroupId eventGroupId;
+  private Set<Integer> operatorsToDeploy = Sets.newHashSet();
+  private Set<Integer> operatorsToUndeploy = Sets.newHashSet();
+  private Set<String> affectedContainers = Sets.newHashSet();
+
+  public DeployRequest()
+  {
+    eventGroupId = EventGroupId.newEventGroupId();
+  }
+
+  public DeployRequest(EventGroupId groupId)
+  {
+    this.eventGroupId = groupId;
+  }
+
+  public EventGroupId getEventGroupId()
+  {
+    return eventGroupId;
+  }
+
+  public Set<Integer> getOperatorsToDeploy()
+  {
+    return operatorsToDeploy;
+  }
+
+  public Set<Integer> getOperatorsToUndeploy()
+  {
+    return operatorsToUndeploy;
+  }
+
+  public Set<String> getAffectedContainers()
+  {
+    return affectedContainers;
+  }
+
+  public void addOperatorToDeploy(int operatorId)
+  {
+    operatorsToDeploy.add(operatorId);
+  }
+
+  public void removeOperatorToDeploy(int operatorId)
+  {
+    operatorsToDeploy.remove(operatorId);
+  }
+
+  public void addOperatorToUndeploy(int operatorId)
+  {
+    operatorsToUndeploy.add(operatorId);
+  }
+
+  public void addAffectedContainer(String containerId)
+  {
+    affectedContainers.add(containerId);
+  }
+
+  public static class EventGroupId extends AbstractWritableAdapter
+  {
+    private static final long serialVersionUID = 1L;
+    public long groupId;
+
+    public static EventGroupId newEventGroupId()
```

Comment:
To address @vrozov 's concern of uniqueness you can consider using operator id or containerid as the prefix. This id is supplied by the caller (assuming all such calls originate in the context of a container or an operator). With such prefixing there will never be accidental collisions.